virttest/libvirt_xml/capability_xml.py

```diff
@@ -118,6 +118,20 @@ def get_guest_capabilities(self):
                     dom_prop = {}
                     if domain.find('emulator') is not None:
                         dom_prop['emulator'] = domain.find('emulator').text
+                        dom_m_list = []
```

Comment:
From your output it is clear that qemu does not have anything ```<machine maxCpus='1'>40p</machine>
      <domain type='qemu'/>
      <domain type='kvm'>
``` looks like system has a local qemu installed which caused a different domcapabilities layout, uninstall local qemu and check for output.
This patch manipulates the dict which inturn is not required, we just need a exact layout of domcapbilities output,
just that tests need to handle in a right way.

Reply:
Thank you @sathnaga for correcting me, I will close this PR.